src/geotools/src/main/java/org/locationtech/geogig/geotools/plumbing/DataStoreImportOp.java

```diff
@@ -0,0 +1,168 @@
+/* Copyright (c) 2016 Boundless and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/edl-v10.html
+ *
+ * Contributors:
+ * Erik Merkle (Boundless) - initial implementation
+ */
+package org.locationtech.geogig.geotools.plumbing;
+
+import org.geotools.data.DataStore;
+import org.locationtech.geogig.api.AbstractGeoGigOp;
+import org.locationtech.geogig.api.GeogigTransaction;
+import org.locationtech.geogig.api.RevCommit;
+import org.locationtech.geogig.api.RevTree;
+import org.locationtech.geogig.api.plumbing.TransactionBegin;
+import org.locationtech.geogig.api.porcelain.AddOp;
+import org.locationtech.geogig.api.porcelain.CommitOp;
+import org.locationtech.geogig.repository.WorkingTree;
+
+import com.google.common.base.Supplier;
+
+/**
+ * Imports feature tress (layers) into a repository from a GeoTools {@link DataStore}.
+ *
+ */
+public class DataStoreImportOp extends AbstractGeoGigOp<RevCommit> {
+
+    private Supplier<DataStore> dataStoreSupplier;
+    // commit options
+    private String authorEmail;
+    private String authorName;
+    private String commitMessage;
+    // import options
+    private String table;
+    private boolean all;
+    private boolean add;
+    private boolean alter;
+    private boolean forceFeatureType;
+    private String dest;
+    private String fidAttribute;
+
+    /**
+     * Set the source {@link DataStore DataStore}, from which features should be imported.
+     *
+     * @param dataStore A {@link com.google.common.base.Supplier Supplier} for the source
+     *                  {@link DataStore DataStore} containing features to import into the
+     *                  repository.
+     * @return A reference to this Operation.
+     */
+    public DataStoreImportOp setDataStore(Supplier<DataStore> dataStore) {
+        this.dataStoreSupplier = dataStore;
+        return this;
+    }
+
+    @Override
+    protected RevCommit _call() {
+
+        final DataStore dataStore = dataStoreSupplier.get();
+
+        RevCommit revCommit;
+        /**
+         * Import needs to: 1) Begin a transaction 2) Import the data 3) Add changes to be staged 4)
+         * Commit staged changes 5) End the transaction
+         */
+        try {
+            // start a transaction
+            final GeogigTransaction transactionCtx = command(TransactionBegin.class).call();
```

Comment:
transactionCtx should be created outside the try block, and transactionCtx.abort() be called if an exception happens